lib/xcodeproj/project/object/native_target.rb

```diff
@@ -416,6 +416,19 @@ class PBXNativeTarget < AbstractTarget
         # @!group Helpers
         #--------------------------------------#
 
+        # Checks product_type to determine whether or not the recevier is a
+        # an app extension
+        #
+        # @return True if the target would be an extension embedded in an
+        #         app target, otherwise false
+        #
+        # @note watchOS 2 extensions are extensions of a watch target, not
+        #       an app target, whereas this is not the case for watchOS 1
+        def app_extension?
```

Comment:
I agree with @segiddins 

@benasher44 you're correct that watchOS 2 extensions need their frameworks embedded, because the binaries are build for a different architecture.